crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -394,7 +395,13 @@ fn execute_move_publish<E: fmt::Debug, S: StorageView<E>>(
             "init should not have return values"
         )
     }
-    Ok(())
+
+    Ok(Value::Object(ObjectValue::new(
+        UpgradeCap::type_(),
+        /* has_public_transfer */ true,
+        /* used_in_non_entry_move_call */ false,
+        &bcs::to_bytes(&UpgradeCap::new(context.fresh_id()?, package_id)).unwrap(),
+    )?))
```

Comment:
I am very glad to see this was easy to do! (I hope! lol)

Reply:
It was!